src/S3/Result/AwsObject.php

```diff
@@ -37,21 +37,21 @@ class AwsObject
 
     /**
      * @param array{
-     *   Key?: string,
-     *   LastModified?: \DateTimeInterface|string,
-     *   ETag?: string,
-     *   Size?: string,
-     *   StorageClass?: string,
-     *   Owner?: \AsyncAws\S3\Result\Owner|array,
+     *   Key: ?string,
+     *   LastModified: ?\DateTimeInterface,
+     *   ETag: ?string,
```

Comment:
Im not sure this is true. We are not sure that all keys always exists.

Reply:
this class is used for output/result. this class is called from the a code generated all keys are hard coded and should exist (value can be null, but all keys exists). If it's not the case, psalm will yield.

Reply:
Okey. 

As long as we know that the response from AWS not always contain all keys.

Reply:
whether or not the AWS reponse contains all keys, this class is instanciated in with all key
in https://github.com/async-aws/aws/blob/4c7eb2d7a772c6f0070d88f357f793fab28edc68/src/S3/Result/ListObjectsV2Output.php#L214-L222

Reply:
I saw that now when doing a second review. 

So the `*Output` classes will do all the checking for null. 
:+1: